integration/test_ft_metadata_cluster_validation.py

```diff
@@ -0,0 +1,283 @@
+            waiters.wait_for_equal(
+                lambda node=node: self.is_indexing_complete_on_node(node, index_name),
+                True,
+                timeout=timeout
+            )
+
+    def get_ft_list_from_all_nodes(self) -> List[List[bytes]]:
+        """Get FT._LIST results from all cluster nodes."""
+        results = []
+        for i in range(self.CLUSTER_SIZE):
+            node = self.new_client_for_primary(i)
+            result = node.execute_command("FT._LIST")
+            results.append(result)
+        return results
+
+    def get_ft_info_from_all_nodes(self, index_name: str) -> List[FTInfoParser]:
+        """Get FT.INFO results from all cluster nodes."""
+        results = []
+        for i in range(self.CLUSTER_SIZE):
+            node = self.new_client_for_primary(i)
+            try:
+                info_response = node.execute_command("FT.INFO", index_name)
+                parser = FTInfoParser(info_response)
+                results.append(parser)
+            except ResponseError as e:
+                # If index doesn't exist on this node, we still want to track this
+                results.append(None)
+        return results
+
+    def validate_ft_list_consistency(self, expected_indexes: List[str]):
+        """Validate that FT._LIST returns consistent results across all nodes."""
+        ft_list_results = self.get_ft_list_from_all_nodes()
+        
+        # Convert all results to sets of strings for comparison
+        normalized_results = []
+        for result in ft_list_results:
+            normalized = set()
+            for item in result:
+                if isinstance(item, bytes):
+                    normalized.add(item.decode('utf-8'))
+                else:
+                    normalized.add(str(item))
+            normalized_results.append(normalized)
+        
+        # All nodes should have the same set of indexes
+        first_result = normalized_results[0]
+        for i, result in enumerate(normalized_results[1:], 1):
+            assert result == first_result, f"Node {i} FT._LIST result differs from node 0: {result} vs {first_result}"
+        
+        # Verify expected indexes are present
+        expected_set = set(expected_indexes)
+        assert expected_set.issubset(first_result), f"Expected indexes {expected_set} not found in {first_result}"
+
+    def validate_ft_info_consistency(self, index_name: str, expected_attributes: Dict[str, Any]):
+        """Validate that FT.INFO returns consistent results across all nodes."""
+        ft_info_results = self.get_ft_info_from_all_nodes(index_name)
+        
+        # All nodes should have the index
+        for i, parser in enumerate(ft_info_results):
+            assert parser is not None, f"Index '{index_name}' not found on node {i}"
+        
+        # Compare key metadata fields across all nodes
+        first_parser = ft_info_results[0]
+        
+        for i, parser in enumerate(ft_info_results[1:], 1):
```

Comment:
I think it would be good to simplify the logic below. What causes `parser.to_dict() == first_parser.to_dict()` to fail?